jpo-asn-runtime/src/main/java/us/dot/its/jpo/asn/runtime/types/Asn1OctetString.java

```diff
@@ -0,0 +1,85 @@
+package us.dot.its.jpo.asn.runtime.types;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonValue;
+import java.util.HexFormat;
+import lombok.Getter;
+
+public class Asn1OctetString implements Asn1Type {
+
+    @Getter
+    protected final int minLength;
+
+    @Getter
+    protected final int maxLength;
+
+    @JsonIgnore
+    protected byte[] octets;
+
+    @JsonValue
+    public String getValue() {
+        HexFormat hexFormat = HexFormat.of().withUpperCase();
+        if (octets != null) {
+            return hexFormat.formatHex(octets);
+        } else {
+            return "";
+        }
+    }
+
+    public void setValue(String value) {
+        try {
+            this.octets = validate(value);
+        } catch (Exception e) {
+            throw new IllegalArgumentException(
+                    String.format("Hex String '%s' is not valid", value), e);
+        }
+    }

```

Comment:
suggestion(non-blocking): separate the validation logic from the construction logic
```suggestion
    public void setValue(String value) {
        try {
            validate(value);
            HexFormat hexFormat = HexFormat.of();
            this.octets = hexFormat.parseHex(value);
        } catch (Exception e) {
            throw new IllegalArgumentException(
                    String.format("Hex String '%s' is not valid", value), e);
        }
    }
```

with 

```java
    protected void validate(String aValue) throws Exception {
        if (aValue == null) {
            throw new Exception("hex string is null");
        }
        // Size of hex format string can be 2 * byte size
        boolean validLength = aValue.length() >= 2 * minLength && aValue.length() <= 2 * maxLength;
        if (!validLength) {
            throw new Exception(String.format("Hex string length out of bounds, %s, number of bytes must be between" +
                    " %d and %d", aValue, minLength, maxLength));
        }
    }
```

Reply:
OK changed that as suggested